lib/LaTeXML/Package/LaTeX.pool.ltxml

```diff
@@ -1367,42 +1367,41 @@ DefEnvironment('{flushright}', sub {
 
 # These add an operation to be carried out on the current node & following siblings, when the current group ends.
 # These operators will add alignment (class) attributes to each "line" in the current block.
-#DefPrimitiveI('\centering',   undef, sub { UnshiftValue(beforeAfterGroup=>T_CS('\@add@centering')); });
-# NOTE: THere's a problem here.  The current method seems to work right for these operators
-# appearing within the typical environments.  HOWEVER, it doesn't work for a simple \bgroup or \begingroup!!!
-# (they don't create a node! or even a whatsit!)
-DefConstructorI('\centering', undef,
-  sub { AssignValue(ALIGNING_NODE => $_[0]->getElement); return; },
+sub setupAligningContext {
+  my ($document) = @_;
+  my $node = $document->getElement;
+  AssignValue(ALIGNING_NODE => [$node, $node->lastChild]);
+  return; }
+
+sub applyAligningContext {
+  my ($document, $align, $class) = @_;
+  if (my $container = LookupValue('ALIGNING_NODE')) {
+    my ($node, $previous) = @$container;
+    my @children = $node->childNodes($node);
```

Comment:
`childNodes()` doesn't take arguments right? Likely has no effect, but threw me off.

Reply:
A related detail is that the previous code used `$document->getChildElements($node)` instead of `childNodes`, the difference being whether text nodes are excluded/included.

`setAlignOrClass` executes `setAttribute` on the input node, so I think it only expects element nodes.

Reply:
Extra arg has no effect for now, but will be confusing later.
We do want text nodes here, since that might be the marker node where we start marking. But you're right that we need to protect `setAlignOrClass`.